Mathlib/RingTheory/Ideal/Span.lean

```diff
@@ -230,21 +214,76 @@ theorem mem_span_insert' {s : Set α} {x y} : x ∈ span (insert y s) ↔ ∃ a,
   Submodule.mem_span_insert'
 
 @[simp]
-theorem span_singleton_neg (x : α) : (span {-x} : Ideal α) = span {x} := by
+theorem span_singleton_neg (x : α) : span {-x} = span {x} := by
   ext
   simp only [mem_span_singleton']
   exact ⟨fun ⟨y, h⟩ => ⟨-y, h ▸ neg_mul_comm y x⟩, fun ⟨y, h⟩ => ⟨-y, h ▸ neg_mul_neg y x⟩⟩
 
 @[simp]
-theorem span_singleton_abs [LinearOrder α] (x : α) :
-    span {|x|} = span {x} := by
+theorem span_pair_neg_left (x y : α) : span {-x, y} = span {x, y} := by
+  rw [span_insert, span_singleton_neg, span_insert]
```

Comment:
I think one of these generalizes to `span (insert (-x) s) = span (insert x s)`

Reply:
I think maybe what you actually want is a `simproc` that simplifies `span _ {a, b, c}` by rewriting it to the `sup` of the spans of the singletons, simplifying each piece, then rewriting back.

Reply:
Or maybe it's even possible to use pre- and post- simp lemmas to achieve this?

Reply:
Yeah, that would be perfect! Unfortunately I don't know how to write one...